src/test/java/StringManipulationTest.java

```diff
@@ -19,130 +38,275 @@ public void tearDown() {
     @Test
     public void testRemoveNthCharacter6() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Remove all characters from this string please and thank you.");
+        String testString = manipulatedstring.removeNthCharacter(1, false);
+        assertEquals("",testString);
     }
 
+    // Test to verify that method will throw a NullPtrException if a string is not set.
     @Test
     public void testRemoveNthCharacter7() {
-        fail("Not yet implemented");
+        assertThrows(NullPointerException.class, ()-> {
+                    Objects.requireNonNull(manipulatedstring).removeNthCharacter(1, true);});
+
     }
 
+    //Initial test with base requirements for getSubStrings.
     @Test
     public void testGeSubStrings1() {
         manipulatedstring.setString("This is my string");
         String [] sStings = manipulatedstring.getSubStrings(3, 4);
-
         assertEquals(sStings[0], "my");
         assertEquals(sStings[1], "string");
     }
 
+    // Test to check if getSubStrings method will throw an IllegalArgumentException with invalid startWord.
     @Test
     public void testGeSubStrings2() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This is a test string.");
+        assertThrows(IllegalArgumentException.class, ()->
+        {manipulatedstring.getSubStrings(-1, 2);});
+
+        assertThrows(IllegalArgumentException.class, ()->
+        {manipulatedstring.getSubStrings(0, 2);});
     }
+
+
+    // Verify getSubStrings will throw an IllegalArgumentException when endWord <=0
     @Test
     public void testGeSubStrings3() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This is a test string.");
+        assertThrows(IllegalArgumentException.class, ()->
+        {manipulatedstring.getSubStrings(1,-1);});
+
+        assertThrows(IllegalArgumentException.class, ()->
+        {manipulatedstring.getSubStrings(1, 0);});
     }
+
+    //Test to verify that getSubStrings will throw an IllegalArgumentException
+    // if startWord > endWord while both are valid inputs.
     @Test
     public void testGeSubStrings4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This is a test string.");
+        assertThrows(IllegalArgumentException.class, ()->
+        {manipulatedstring.getSubStrings(4, 2);});
     }
+
+    // This test is to verify getSubStrings will return a single (and correct) word
+    // when startWord = endWord and the String array is the correct length.
     @Test
     public void testGeSubStrings5() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("The word \'correct\' is the correct word.");
+        String[] testString = manipulatedstring.getSubStrings(6, 6);
+        String[] actualString = {"correct"};
+        int length = testString.length;
+        assertEquals(1, length);
+        assertEquals(actualString[0], testString[0]);
```

Comment:
I pretty much don't have any reviews to be done. For my final review's sake. I would like to point out that assertEquals for testing is usually expected to actual. so by ethics its expectedString[0].